src/main/scala/me/rexim/morganey/module/Module.scala

```diff
@@ -1,6 +1,15 @@
 package me.rexim.morganey.module
 
-class Module(resourcePath: String) {
-  def name: String =
-    resourcePath.replace('/', '.').stripSuffix(s".${ModuleFinder.fileExtension}")
+import me.rexim.morganey.ast.MorganeyBinding
+import scala.util._
+
+class Module(modulePath: ModulePath, classLoader: ClassLoader = Module.getClass.getClassLoader) {
+  def canonicalPath: String =
+    modulePath.asCanonicalPath.path
+
+  // TODO: Implement Module.bindings
+  def bindings: Try[Set[MorganeyBinding]] = ???
+
+  // TODO: Implement Module.dependencies
```

Comment:
I think these two subtasks can be merged into a single one. Their scope feels pretty small.